### ISSUE TITLE
MagiskManager Fails to Load on LineageOS 16 Unofficial After Using Hide MagiskManager (Oneplus3)

### ISSUE THREAD

**Issue by Alaric323** [topjohnwu/Magisk#655]
Hey topjohnwu,

Having an issue where installing Magisk 17.2 to an unofficial build of LineageOS causes the system to act glitchy, such as apps like play store crashing.  Magisk manager doesn't load to the main screen (just sits at the icon window forever until the system tries to kill it) and root is not available.

Magisk 17.1, however, works fine as far as I can tell.

Sorry I can't get any logs, as magisk 17.2 doesn't save to the MagiskManager folder anymore to avoid some rude root detection. :P

**Comment by Didgeridoohan** [comment]
You've only halfway done your homework... When you save a log the Manager tells you with a flash window where it's saved. Hint: Downloads.

**Comment by Alaric323** [comment]
One slight problem with that...  If you can't get into the manager, you can't save a log.

**Comment by Didgeridoohan** [comment]
Just pull the Magisk log from /cache then... Or save a logcat. There are a whole number of ways to get logs.

But, it's a moot point anyway because you've not followed the instructions on how to report bugs:
https://github.com/topjohnwu/Magisk/blob/master/README.MD#bug-reports

**Comment by Alaric323** [comment]
Right then, after upgrading to the latest Canary build (17.3-b127e018), I've found that this issue only happens after enabling "Hide MagiskManager", in other words, rebuilding the package with a random identifier.  The regular MagiskManager bundled with the installer works fine.

I'll work on getting the logs off now.

**Comment by Alaric323** [comment]
So interesting thing happened when I reinstalled Magisk Manager with the renamed Manager installed still.  The renamed Manager asked for root permissions from the main app.  Makes me think that the renamed Magisk isn't hooking to the su binary in some way?

**Comment by Didgeridoohan** [comment]
The repackaged Manager's new package name is likely not being saved to the Magisk db properly. Therefore it won't get root access... It's a safety measure: If the package name isn't in the Magisk db, it's not the proper Magisk Manager.

**Comment by topjohnwu** [comment]
@Alaric323 try uninstall your current Magisk Manager, install the official APK, then try to repackage from that. This doesn't seem to be an issue with Magisk itself but rather some weird issues happening in the database.

**Comment by Alaric323** [comment]
Unfortunately, doing as @topjohnwu suggested has not solved anything.  I uninstalled any versions of Magisk on the system, rebooted, and the system automatically loaded the "lite" version of magisk that asks to do the upgrade to the full version.

I ignored that, and installed the latest V6.0.0 version from the APK.  Hiding Magisk from there results in the same problem as earlier, and upon reboot, Magisk reloads the lite version.
